File: scan/src/main/java/com/dreampany/scan/data/source/local/Converters.java
package com.dreampany.scan.data.source.local;

import androidx.room.TypeConverter;

import com.dreampany.scan.data.enums.ScanType;

/**
 * Created by Hawladar Roman on 15/6/18.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 */
public class Converters {
    @TypeConverter
    synchronized public static int fromScanType(ScanType type) {
        return type.ordinal();
    }

    @TypeConverter
    synchronized public static ScanType toScanType(int code) {
        return ScanType.valueOf(code);
    }
}

File: tools/src/main/java/com/dreampany/tools/data/source/crypto/room/converters/Converters.kt
package com.dreampany.tools.data.source.crypto.room.converters

import androidx.room.TypeConverter
import com.dreampany.framework.data.source.room.converter.Converter
import com.dreampany.tools.data.enums.crypto.Currency

/**
 * Created by roman on 3/19/20
 * Copyright (c) 2020 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class Converters : Converter() {

    @Synchronized
    @TypeConverter
    fun toString(input: Currency?): String? {
        return if (input == null) null else input.name
    }

    @Synchronized
    @TypeConverter
    fun toCurrency(input: String?): Currency? {
        return if (input.isNullOrEmpty()) null else Currency.valueOf(input)
    }
}